src/pages/index.js

```diff
@@ -76,6 +122,36 @@ function closeModal(modal) {
   modal.classList.remove("modal_opened");
   document.removeEventListener("keydown", handleModalEsc);
 }
+let selectedCard;
+let selectedCardId;
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteCardModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const deleteCardModalBtn = evt.submitter;
+  setButtonText(deleteCardModalBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then((data) => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      selectedCard = "";
```

Comment:
```js

      selectedCard = "";
      selectedCardId = "";
```

These 2 lines of code should be in `.then` in line 141 not to clear the data if there is an error

And they should be `undefined`, because you don't define them in lines 125 and 126


```js
    .then((data) => {
      selectedCard.remove();
      closeModal(deleteCardModal);
      selectedCard = undefined;  // here
      selectedCardId = undefined; // here
    })
    .catch((err) => {
      console.error(err);
    })
    .finally(() => {
      setButtonText(deleteCardModalBtn, false, "Delete", "Deleting");
    });
```